src/passes/OptimizeInstructions.cpp

```diff
@@ -502,6 +502,10 @@ struct OptimizeInstructions
         // math operations on a constant power of 2 right side can be optimized
         if (right->type == Type::i32) {
           uint32_t c = right->value.geti32();
+          if (!(c >> 31) && isSignedBinaryOp(binary->op) &&
```

Comment:
I think this would be clearer if the `!(c >> 31)` were replaced with a cast to a signed type and a comparison to 0.

Reply:
Done!